packages/oauth-adapters/src/oauthAuthenticationAdapter.ts

```diff
@@ -1,66 +1,94 @@
-import { OAuthToken } from './oAuthToken';
 import {
   AuthenticatorInterface,
   passThroughInterceptor,
 } from '@apimatic/core-interfaces';
 import { AUTHORIZATION_HEADER, setHeader } from '@apimatic/http-headers';
 import { OAuthConfiguration } from './oAuthConfiguration';
 
-export const requestAuthenticationProvider = (
-  initialOAuthToken?: OAuthToken,
-  oAuthTokenProvider?: (token: OAuthToken | undefined) => Promise<OAuthToken>,
-  oAuthOnTokenUpdate?: (token: OAuthToken) => void,
-  oAuthConfiguration?: OAuthConfiguration
+interface OAuthTokenConstraints {
+  accessToken?: string;
+  expiry?: bigint;
+}
+
+async function refreshOAuthToken<T extends OAuthTokenConstraints>(
+  currentToken: T | undefined,
+  provider: (token: T | undefined) => Promise<T>,
+  onUpdate?: (token: T) => void,
+  clockSkew?: number
+): Promise<T | undefined> {
+  if (
+    !provider ||
+    (isValid(currentToken) && !isExpired(currentToken as T, clockSkew))
```

Comment:
The casting to the generic type T is unnecessary here and can be safely removed. Additionally, please adjust the file structure to follow best practices by moving the exported function to the top of the file for better readability and maintainability.